src/swaps/fastbtc/FastbtcSwapProvider.js

```diff
@@ -0,0 +1,254 @@
+      return mapValues(totalFees, f => unitToCurrency(cryptoassets[asset], f))
+    }
+
+    if (txType in FastbtcSwapProvider.feeUnits) {
+      const fees = {}
+      for (const feePrice of feePrices) {
+        fees[feePrice] = getTxFee(FastbtcSwapProvider.feeUnits[txType], asset, feePrice)
+      }
+      return fees
+    }
+  }
+
+  async waitForSendConfirmations ({ swap, network, walletId }) {
+    const client = this.getClient(network, walletId, swap.from, swap.fromAccountId)
+
+    try {
+      const tx = await client.chain.getTransactionByHash(swap.swapTxHash)
+      if (tx && tx.confirmations > 0) {
+        return {
+          endTime: Date.now(),
+          status: 'WAITING_FOR_RECEIVE'
+        }
+      }
+    } catch (e) {
+      if (e.name === 'TxNotFoundError') console.warn(e)
+      else throw e
+    }
+  }
+
+  async waitForReceive ({ swap, network, walletId }) {
+    try {
+      const toChain = cryptoassets[swap.to].chain
+      const toAddressRaw = await this.getSwapAddress(network, walletId, swap.to, swap.toAccountId)
+      const toAddress = chains[toChain].formatAddress(toAddressRaw)
+      const addressHistory = (await this._getHistory(toAddress)).sort((a, b) => new Date(a.dateAdded).getTime() > new Date(b.dateAdded).getTime() ? 1 : -1)
+      let isDepositConfirmed = false
+      let isReceiveConfirmed = false
+      let depositConfirmationDate = 0
+      let depositAmount = 0
+      for (const transaction of addressHistory) {
+        if (transaction.txHash === swap.swapTxHash &&
+          transaction.status === 'confirmed' && transaction.type === 'deposit') {
+          isDepositConfirmed = true
+          depositConfirmationDate = new Date(transaction.dateAdded).getTime()
+          depositAmount = transaction.valueBtc
+        } else if (isDepositConfirmed && transaction.status === 'confirmed' && transaction.type === 'transfer' &&
+          transaction.valueBtc === depositAmount && new Date(transaction.dateAdded).getTime() - depositConfirmationDate > 0 && new Date(transaction.dateAdded).getTime() - depositConfirmationDate < 86400000) {
+          isReceiveConfirmed = true
+        }
+      }
+      if (isDepositConfirmed && isReceiveConfirmed) {
+        return {
+          endTime: Date.now(),
+          status: 'SUCCESS'
+        }
+      }
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  async performNextSwapAction (store, { network, walletId, swap }) {
+    let updates
+
+    switch (swap.status) {
+      case 'WAITING_FOR_SEND_CONFIRMATIONS':
+        updates = await withInterval(async () => this.waitForSendConfirmations({ swap, network, walletId }))
+        break
+      case 'WAITING_FOR_RECEIVE':
+        updates = await withInterval(async () => this.waitForReceive({ swap, network, walletId }))
+        break
+    }
+
+    return updates
+  }
+
+  static txTypes = {
+    SWAP_INITIATION: 'SWAP_INITIATION'
+  }
+
+  static feeUnits = {
+    SWAP_INITIATION: {
+      BTC: 290,
```

Comment:
is feeUnits even required here at all? BTC is calculated dynamically, and you can only go BTC -> RBTC and you don't need to pay fees for RBTC. So only estimation for BTC is required